### PR TITLE
Use go-build for commands and add a workspace

### PR DESCRIPTION
**Pull request by anuraaga**
Initially had copied the build tooling for `wasilibs/tools` as with other projects in the org, but this isn't actually a Wasm program but a normal Go package. So I have a different better package (and account...) for that :) The biggest difference is handling md / yaml formatting too.

### CODE REVIEW

**Review by ncruces — APPROVED** [review]
OK! Just noticed, we should probably keep the “Round up to the page” comment similar on Windows, unless there's a specific reason not to.

File: allocator/nonmoving_test.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ func TestNonMoving(t *testing.T) {
 		{
 			name: "native",
 			mem:  NewNonMoving().Allocate(10, 20),
-			cap:  int(pageSize),
+			cap:  int(pageSize), //nolint:unconvert // depending on platform, not an int
```
Comment (ncruces) [inline allocator/nonmoving_test.go]: Is this needed? It can probably be an untyped constant on Windows, and an `int` elsewhere.

Comment (anuraaga) [reply allocator/nonmoving_test.go]: Unless there's some technique I don't know, a `const foo = 12414` is always an `int` (a literal on the other hand is implicitly typed by the expression). So I went with using `int` on windows like elsewhere and convert to `uint64` in the logic which seems fine

Comment (ncruces) [reply allocator/nonmoving_test.go]: Constants in Go are actually untyped, until you coerce them. Untyped integer constants just default to `int`, but they can be used in any integer context without conversion.

https://go.dev/blog/constants

**Pull request merged**